src/main/java/org/folio/service/processing/split/FileSplitService.java

```diff
@@ -114,18 +103,29 @@ public Future<CompositeFuture> splitStream(
       .onComplete(ar1 -> LOGGER.info("File split for key={} completed", key));
 
     return promise
+      // original future resolves once the chunks are split, but NOT uploaded
       .future()
-      .onSuccess(cf ->
-        cf.onSuccess(innerResult -> {
-          LOGGER.info("Deleting temporary folder={}", tempDir);
-
-          try {
-            Files.delete(tempDir);
-          } catch (IOException e) {
-            // not severe enough to fail the whole upload, though
-            LOGGER.error("Could not delete temporary folder", e);
-          }
-        })
-      );
+      // this composite future resolves once all are uploaded
+      .compose(cf -> cf)
+      // now let's turn this back into a List<String>
+      .map(cf -> cf.list())
+      .map(list ->
+        list.stream().map(String.class::cast).collect(Collectors.toList())
+      )
+      // and since we're all done, we can delete the temporary folder
+      .map((List<String> innerResult) -> {
```

Comment:
nitpick: seems like compose would be appropriate instead of map. Only because of  business logic to delete folders.

Reply:
The Java File API is synchronous; compose is for futures/async logic

Reply:
ah, thought this was vertx.